roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -21,13 +21,45 @@
       openshift_hosted_registry_storage_s3_cloudfront_keypairid and
       openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile are required
 
+
 # Copy the cloudfront.pem to the host if the baseurl is given
-- name: Copy cloudfront.pem to the registry
-  copy:
-    src: "{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile }}"
-    dest: /etc/s3-cloudfront/cloudfront.pem
-    backup: true
-    owner: root
-    group: root
-    mode: 0600
+- block:
+
+    - set_fact:
+        s3_cloudfront_config_dirname: "/etc/origin"
+        s3_cloudfront_target_volume: "cloudfront-vol"
+
+    - set_fact:
+        s3_cloudfront_remote_privkey_path: "{{ s3_cloudfront_config_dirname }}/{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile | basename }}"
+
+    - name: Ensure s3-cloudfront directory exists
+      file:
+        path: "{{ s3_cloudfront_config_dirname }}"
+        state: directory
+        owner: root
+        group: root
+
+    - name: Copy cloudfront.pem to the registry
+      copy:
+        src: "{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile }}"
+        dest: "{{ s3_cloudfront_remote_privkey_path }}"
+        backup: true
+        owner: root
+        group: root
+        mode: 0600
+
+    - name: Create registry secrets for cloudfront private key file
+      oc_secret:
+        state: present
+        namespace: openshift-infra
+        name: cloudfront
+        files:
+          - name: cloudfront
+            path: "{{ s3_cloudfront_remote_privkey_path }}"
```

Comment:
In theory ``failed_when`` should be able to used as a list but I'll double check (Ref: https://github.com/ansible/ansible/pull/14833)